Chummer/customdata/manifest.xsd

```diff
@@ -0,0 +1,39 @@
+<?xml version="1.0" encoding="utf-8"?>
+<xs:schema id="manifest"
+    targetNamespace="http://tempuri.org/manifest.xsd"
+    elementFormDefault="qualified"
+    xmlns="http://tempuri.org/manifest.xsd"
+    xmlns:mstns="http://tempuri.org/manifest.xsd"
+    xmlns:xs="http://www.w3.org/2001/XMLSchema"
+>
+  <xs:complexType name ="Manifest">
+    <xs:sequence>
+      <xs:element name="guid" type="xs:string" minOccurs="1" maxOccurs="1"/>
+      <xs:element name="name" type="xs:string" minOccurs="1" maxOccurs="1"/>
+      <xs:element name="version" type="xs:string" minOccurs="0" maxOccurs="1"/>
+      <xs:element name="author" type="xs:string" minOccurs="0" maxOccurs="1"/>
+      <xs:element name="descriptions" minOccurs="0" maxOccurs="1">
+        <xs:complexType>
+          <xs:sequence>
+            <xs:element name="description" minOccurs="0" maxOccurs="unbounded"/>
+          </xs:sequence>
+        </xs:complexType>
+      </xs:element>
+      <!--Those don't have functionality yet, the idea for later is to use GuID to identify dependencies and exclusivities, maybe with the anyof/allof schema-->
```

Comment:
Dependencies would probably need to work off of both an ID and a version number, not just an ID.

They could also work off of a hash from the contents of the other ruleset(s), but I already commented about how I have no idea how I would integrate that into the manifest. Even if it's calculated only within Chummer without the user needing to put it inside their manifest file, there's still the problem of how do you get the hashes of other rulesets into your dependency info for your own manifest without opening up Chummer and checking manually...

Reply:
Chummer could calculate the hash and add it to the `.xml` as a `hash /` node the first time a custom data directory is loaded and no hash info is present. I don't know how resource hungry calculating hashes is, we could just calculate it each time. Then the user won't have to delete the hash each time he changed his file.

If a user now wants to create it's own dependency, they would just need to copy out the hash from one file into the next.
Copying over the GUID and Version or the auto generated hash, are probably equivalent in complexity. 

But we should then define dependencies with `<name />` and `<hash />`. The name is just needed for human readability. Maybe including the `<version />` for this purpose as well would be a good idea.

Reply:
Creating a SHA512 hash/checksum from the contents of files should take milliseconds unless the files in total are larger than about 100 MB. The difficulty is more from the setup of needing to launch Chummer in order get the hash in the first place. Even if it's done automatically, you'd still have to launch Chummer each time you changed something to get an updated hash for your custom ruleset.

Reply:
With hashes wo won't get around that.
We could throw in some additional work to make it more intuitive, to start up chummer for doing stuff like this.

For example a way do create dependencies in chummer.

- Open character files.
- Select a Custom Data.
- There is a add dependencies button.
- You select another directory.
- Chummer does all the background of adding the hash to the correct file.

If we not only use a hash as identifier, we could also update the hash in all other files, that are identified with name and (older) version. That where dependend upon the chabged file.
This would  make updating a data file really easy. As long as you are sure, that you didn't disrupt any dependencies.